src/client/pfs/pfs.proto

```diff
@@ -340,23 +291,27 @@ service InternalAPI {
 
   // Commit rpcs
   // StartCommit creates a new write commit from a parent commit.
-  rpc StartCommit(StartCommitRequest) returns (google.protobuf.Empty) {}
+  rpc StartCommit(StartCommitRequest) returns (Commit) {}
+  // Fork creates a commit on a new branch.
+  rpc Fork(ForkRequest) returns (Commit) {}
   // FinishCommit turns a write commit into a read commit.
   rpc FinishCommit(FinishCommitRequest) returns (google.protobuf.Empty) {}
-  // ArchiveCommit marks a commit as archived, it will be excluded from ListCommit.
-  rpc ArchiveCommit(ArchiveCommitRequest) returns (google.protobuf.Empty) {}
+  // ArchiveCommits marks a commit as archived, it will be excluded from ListCommit.
+  rpc ArchiveCommits(ArchiveCommitsRequest) returns (google.protobuf.Empty) {}
```

Comment:
How attached to the "s" are you?
Other requests, namely `FlushCommit` take multiple commits but use a singular which I think makes sense since they can take a single commit. It also just seems nicer to me to have singulars everywhere.